### PR TITLE
Fixes for AEP-131

### LINKED ISSUE #10: Audit AEP-131
Audit the linter rules and docs for AEP-131

### PR DESCRIPTION
**Pull request by rambleraptor**
Fixes #10 

This makes the following changes:

* Makes all changes from `name` -> `path`
* Renames several rules that have `name` in their name
* Changes a couple tests in other rules with the `name` parameter changes.
* Added in Must / Should for most all rules.
* Found a bug with how we handle must / should.

### CODE REVIEW

File: lint/rule_enabled.go [hunk 1]
```diff
@@ -68,7 +68,7 @@ var descriptorDisableChecks = []func(d desc.Descriptor) bool{
 func ruleIsEnabled(rule ProtoRule, d desc.Descriptor, l *dpb.SourceCodeInfo_Location,
 	aliasMap map[string]string, ignoreCommentDisables bool) bool {
 	if(rule.GetRuleType() == MustRule) {
```
Comment (toumorokoshi) [inline lint/rule_enabled.go:70]: Is this bug in the original linter? we should upstream if so.

File: rules/aep0131/request_path_reference.go [hunk 2]
```diff
@@ -20,10 +20,11 @@ import (
...
-var requestNameReference = &lint.FieldRule{
-	Name: lint.NewRuleName(131, "request-name-reference"),
+var requestPathReference = &lint.FieldRule{
+	Name: lint.NewRuleName(131, "request-path-reference"),
 	OnlyIf: func(f *desc.FieldDescriptor) bool {
-		return utils.IsGetRequestMessage(f.GetOwner()) && f.GetName() == "name"
+		return utils.IsGetRequestMessage(f.GetOwner()) && f.GetName() == "path"
 	},
 	LintField: utils.LintFieldResourceReference,
+	RuleType: lint.NewRuleType(lint.ShouldRule),
```
Comment (toumorokoshi) [inline rules/aep0131/request_path_reference.go:29]: not a must? seems like we should annotated this with a resource ref if possible.

**Review by toumorokoshi — APPROVED** [review]
LGTM! one question to consider.

File: lint/rule_enabled.go (hunk 1 above)
Comment (rambleraptor) [reply lint/rule_enabled.go:70]: The Must/Should thing is an addition that I made to our linter only. Should've caught that earlier when I made the change.

File: rules/aep0131/request_path_reference.go (hunk 2 above)
Comment (rambleraptor) [reply rules/aep0131/request_path_reference.go:29]: We specifically changed this to a should in the AEPs. Not sure why.

**Pull request merged**